hide mod engagement

NinaHerrmann opened the pull request:
Hello, 
sorry for the mess and the time wasted!
I put all changed in one commit on this branch with the latest changes of moodle 3.10 version.
For reference those were the goals:
- [x] created course setting for displaying deadline and submissions 
- [x] use the admin settings which are divided for assign lection etc. 
       Remark: I decided to implement that functionality as follows: In case all activities ('assign', 'quiz', 'choice', 'feedback', 'lesson', 'data' ) are deactivated in the admin settings, the course setting is not shown at all. In case at least one of them is activated the setting is shown. The string list all enabled activities so lecturers hopefully do not get confused (lib.php l.705). Another option would be to have an individual setting for each activity also in the course, but I rather considered that as over-engineering as the majority is happy with having it displayed. 
- [x] Fix default is set to 'Yes' when at least one admin configuration is enabled.
- [x] write tests 

This was the feedback/review:

> Can't ~'5d3c540#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R699-R702~  use 'implode' (https://www.php.net/manual/en/function.implode.php) instead?

Thanks, changed to implode :+1: 
https://github.com/NinaHerrmann/moodle-format_topcoll/blob/d8af01bcf20515a97bc7ad2bb917372202807ef3/lib.php#L721
> With 5d3c540#diff-972c351ef4188f1f82cc82207e98147ccbd23e43ddf222a3e5c3f8df5e475d04R120 - better if 'Show additional information for: {$a} in the course';

Thanks, changed that https://github.com/NinaHerrmann/moodle-format_topcoll/blob/d8af01bcf20515a97bc7ad2bb917372202807ef3/lang/en/format_topcoll.php#L120

> As ''Show additional information for {$a} in the course'; - then better English for 5d3c540#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R385-R391 with 5d3c540#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R386 to be 'assignments, quizzes....' etc. And even better if code can be 'assignments, quizzes, lessons and feedbacks' etc. So a trailing 'and' instead of a comma.

Added the `and` :+1: (l.723)
https://github.com/NinaHerrmann/moodle-format_topcoll/blob/d8af01bcf20515a97bc7ad2bb917372202807ef3/lib.php#L723


> https://github.com/gjb2048/moodle-format_topcoll/pull/81/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R538 etc. is invalid as you're getting a site setting for the format that does not exist - i.e. /course/format/topcoll/settings.php.

https://github.com/NinaHerrmann/moodle-format_topcoll/blob/d8af01bcf20515a97bc7ad2bb917372202807ef3/lib.php#L532
I am not sure if I got that one correctly. If the setting is not set the function returns false so I fill it with 2, but correct me if I am wrong!


> Multiple line comments (https://github.com/gjb2048/moodle-format_topcoll/pull/81/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R532-R535) need to use '/*' etc.

https://github.com/NinaHerrmann/moodle-format_topcoll/blob/d8af01bcf20515a97bc7ad2bb917372202807ef3/lib.php#L532
Used multiline comma, code checker complains I can alternatively make it a oneliner

> Don't add a comma when it is not needed: https://github.com/gjb2048/moodle-format_topcoll/pull/81/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R529

implode solves the problem :+1:
https://github.com/NinaHerrmann/moodle-format_topcoll/blob/d8af01bcf20515a97bc7ad2bb917372202807ef3/lib.php#L722

> Here: https://github.com/gjb2048/moodle-format_topcoll/pull/81/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R532-R545 have this logic: https://github.com/gjb2048/moodle-format_topcoll/pull/81/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R755-R758 sort of thing.

Moved this logic to the top of the function so it can be used multiple times https://github.com/NinaHerrmann/moodle-format_topcoll/blob/d8af01bcf20515a97bc7ad2bb917372202807ef3/lib.php#L387

> So only add
> moodle-format_topcoll/lib.php 
```
 $courseformatoptions['showadditionalmoddata'] = array( 
     'default' => $default, 
     'type' => PARAM_INT); 
```
moodle-format_topcoll/lib.php

Should be added. Not sure if you meant that the if should be removed completely but has now the default value.
https://github.com/NinaHerrmann/moodle-format_topcoll/blob/d8af01bcf20515a97bc7ad2bb917372202807ef3/lib.php#L540

> https://github.com/gjb2048/moodle-format_topcoll/pull/81/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R386-R406 needs to be within https://github.com/gjb2048/moodle-format_topcoll/pull/81/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R407

I dont think so since it is used in https://github.com/gjb2048/moodle-format_topcoll/blob/688671921c040e8da297fe07d6d39513370b5392/lib.php#L720 but correct me if I am wrong!

gjb2048 commented:
@NinaHerrmann Ok, please fix https://travis-ci.org/github/gjb2048/moodle-format_topcoll/builds/750995966 and the commit references in https://github.com/gjb2048/moodle-format_topcoll/pull/91#issue-543995731 so that I can have an idea of what you're describing.

gjb2048 commented:
Need better English here: https://github.com/gjb2048/moodle-format_topcoll/pull/91/files#diff-972c351ef4188f1f82cc82207e98147ccbd23e43ddf222a3e5c3f8df5e475d04R121.

gjb2048 commented:
If https://github.com/gjb2048/moodle-format_topcoll/pull/91/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R388 is the 'get' where is the data set?

gjb2048 commented:
Indenting is wrong here: https://github.com/gjb2048/moodle-format_topcoll/pull/91/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R532-R534

NinaHerrmann commented:
> @NinaHerrmann Ok, please fix https://travis-ci.org/github/gjb2048/moodle-format_topcoll/builds/750995966 and the commit references in #91 (comment) so that I can have an idea of what you're describing.

https://github.com/NinaHerrmann/moodle-format_topcoll/blob/d8af01bcf20515a97bc7ad2bb917372202807ef3/lib.php#L532 is the reference now, I hope it is alright if I use permalinks so it can be backtracked to the specific commit.
Just saw that `/* */` is allowed as opposed to 
```
/*
*
*/
```
Changed it to `/* */ `

> Need better English here: https://github.com/gjb2048/moodle-format_topcoll/pull/91/files#diff-972c351ef4188f1f82cc82207e98147ccbd23e43ddf222a3e5c3f8df5e475d04R121. Such as "Allow all users to see the activity deadline and users with grading permission to see the number of submissions on the course page for activities.

https://github.com/NinaHerrmann/moodle-format_topcoll/blob/d8af01bcf20515a97bc7ad2bb917372202807ef3/lang/en/format_topcoll.php#L121
Changed sorry for that, spelling is not my major strength.

> If https://github.com/gjb2048/moodle-format_topcoll/pull/91/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R388 is the 'get' where is the data set?

As far as I can see it is set to the default to 2 when creating the course since settings.php is called.

> Indenting is wrong here: https://github.com/gjb2048/moodle-format_topcoll/pull/91/files#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R532-R534

I think the code checker would complain if the indenting is wrong are you sure?

NinaHerrmann commented:
Updated the first messages so it points to the changes made.

gjb2048 commented:
@NinaHerrmann Ok, will look when I can.

gjb2048 commented:
Ok, https://github.com/gjb2048/moodle-format_topcoll/pull/91/commits/d8af01bcf20515a97bc7ad2bb917372202807ef3#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R532-R534 is still wrong, needs to be:

````
            /* Case 1 : at least one plugin is set to yes show config with default either:
             * Previousvalue - in case it was set.
             * Yes (2) - if no value was set.
             */
````

Sort of thing with consistent indenting.  But what is 'Previousvalue's value when 'Yes' is '2'?  Confused.

gjb2048 commented:
@NinaHerrmann Please don't bump the version number.  This causes agro!

NinaHerrmann commented:
> Ok, [d8af01b#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R532-R534](https://github.com/gjb2048/moodle-format_topcoll/commit/d8af01bcf20515a97bc7ad2bb917372202807ef3#diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395R532-R534) is still wrong, needs to be:
> 
> ```
>             /* Case 1 : at least one plugin is set to yes show config with default either:
>              * Previousvalue - in case it was set.
>              * Yes (2) - if no value was set.
>              */
> ```
> 
> Sort of thing with consistent indenting. But what is 'Previousvalue's value when 'Yes' is '2'? Confused.
As far as I understood 
```
/* Case 1 : at least one plugin is set to yes show config with default either:
* Previousvalue - in case it was set.
* Yes (2) - if no value was set.
*/
 ```
 is not allowed by the code checker so it would be:
 ```
/* Case 1 : at least one plugin is set to yes show config with default either:
    Previousvalue - in case it was set.
    Yes (2) - if no value was set. */
 ```
Correct?
I updated the comment to make clearer what it does. In case the person editing the course set the value previously, this value is taken, otherwise, the default is 2 which stand for yes show additional mod data. I updated variable names and the comment so it hopefully gets clearer. 


> @NinaHerrmann Please don't bump the version number. This causes agro! Please leave that to me to work out.

Sorry about that! don't know how it came in the commit :disappointed:

gjb2048 commented:
Ok, I'm worried about https://github.com/gjb2048/moodle-format_topcoll/pull/91/files#diff-b84ed9a78d3a736009c7632388c0976072bbbafc40ba5b8e3968475a5378accdR224 overwriting the passed in $course - so this is what I'm going to do:

1. Merge in your changes.
2. Add my own 'tidy up' commit on top so that you don't loose credit.

gjb2048 commented:
Reviewd.

NinaHerrmann commented:
@gjb2048 thanks for your support, review, time spent on it, and helpful feedback. And most of all, thanks for finishing it! :)

gjb2048 commented:
@NinaHerrmann No problem.
